### ISSUE TITLE
BUG: When removing line item that hasn't been saved yet using `$order->removeLineItem`.

### ISSUE THREAD

**Issue by samhibberd** [craftcms/commerce#1555]
**Description**
We are running into an issue when trying to remove a specific line item that hasn't been saved yet, when the cart only contains other new items. For us we are actioning this from the `beforeSave` event in a custom order behavior.

Tracked the issue down to the `$order->removeLineItem` method, specifically the following condition:

https://github.com/craftcms/commerce/blob/e3f42a9d7b3e7d1bd16b5e3486aa74ac9597bd5a/src/elements/Order.php#L1525

If you pass a new item to be removed, it will have an id value of `null`, if there are any existing line items (in addition to the item being removed) that are also new then they will also have an id value of `null`.

This means that the first of the conditions will incorrectly return `true` as they are both `null` and incorrectly remove this item.

In this case the second condition `$lineItem === $item` will never get checked.

Would it be safe to assume that if `$item->id` is `null` then the only required check should be `$lineItem === $item`?

```php
if (($item->id !== null && $lineItem->id == $item->id) || $lineItem === $item) {
        unset($lineItems[$key]);
        $this->setLineItems($lineItems);
}
```

**Additional info**
- Craft version: 3.5.0-beta.3
- Commerce version: 3.1.10
- PHP version:
- Database driver & version:
- Other Plugins & versions:

**Comment by lukeholder** [comment]
Thanks for reporting that! I’ve just fixed it for the next release.

To get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/commerce": "dev-develop#bd997657303abde0fed4897e6aef8849ded8baff as 3.1.11",
  "...": "..."
}
```

Then run `composer update`.

**Comment by lukeholder** [comment]
Fixed in bd997657303abde0fed4897e6aef8849ded8baff
